### PR TITLE
fix: onclick handler update in ButtonGroup Widget

### LINKED ISSUE #12410: [Bug]: Button group widget onClick doesn't work for buttons
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Button group widgets on click functionality doesn't work when a button is bound to onClick both in edit and view mode.

### Steps To Reproduce

1. Add a new button group widget
2. Bind onClick property for the buttons in the button group
3. Click the button to see that the action doesn't get executed

### Public Sample App

_No response_

### Version

Production

### PR DESCRIPTION
**Pull request by techbhavin**
## Description

> Button group widget onClick doesn't work for buttons 

Fixes #12410 

## Type of change

- Bugfix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>

    // Code coverage diff between base branch:release and head branch: fix/buttongroup-onclick 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**
 :green_circle: | app/client/src/widgets/ButtonGroupWidget/component/index.tsx | 19.05 **(0.13)** | 8.64 **(0)** | 2.27 **(0.05)** | 20.15 **(0)**</details>

### CODE REVIEW

**Review by yaldram — APPROVED** [review]

**Comment by somangshu** [comment]
Tested the deploy preview and the failing case now works. @appsmithorg/qa can confirm further

**Comment by vivekverma2312** [comment]
@techbhavin Tested On click and it is working as expected.

File: app/client/cypress/fixtures/buttonGroupDsl.json [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                        "label": "More",
+                        "iconName": "more",
+                        "id": "groupButton3",
+                        "buttonType": "SIMPLE",
+                        "placement": "CENTER",
+                        "buttonColor": "#03B365",
+                        "widgetId": "",
+                        "isVisible": true,
+                        "isDisabled": false,
+                        "index": 2,
```
Comment (Tooluloope) [inline app/client/cypress/fixtures/buttonGroupDsl.json:67]: @techbhavin is it possible for a `SIMPLE` ButtonType to have MenuItems?

Comment (Tooluloope) [reply app/client/cypress/fixtures/buttonGroupDsl.json:67]: Also, could you add tests for the other buttons and also the menuItems

**Review by Tooluloope — CHANGES_REQUESTED** [review]

File: app/client/cypress/fixtures/buttonGroupDsl.json (hunk 1 above)
Comment (techbhavin) [reply app/client/cypress/fixtures/buttonGroupDsl.json:67]: @NandanAnantharamu  can you plz look here

Comment (NandanAnantharamu) [reply app/client/cypress/fixtures/buttonGroupDsl.json:67]: @Tooluloope dsl file is auto generated based on my actions, so it did generate based on that and simple is available as option PFA

Comment (Tooluloope) [reply app/client/cypress/fixtures/buttonGroupDsl.json:67]: @techbhavin Should this be possible?

**Comment by techbhavin** [comment]
Here we cover the required test case. so @somangshu  @Tooluloope we can merge this PR.

**Review by Tooluloope — APPROVED** [review]

**Pull request merged**